src/main/java/org/javarosa/core/util/externalizable/ExtWrapArrayListMultiMap.java

```diff
@@ -0,0 +1,93 @@
+package org.javarosa.core.util.externalizable;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+public class ExtWrapArrayListMultiMap extends ExternalizableWrapper {
+
+    private ExternalizableWrapper keyType;
+
+    /* Constructors for serialization */
+
+    public ExtWrapArrayListMultiMap(Multimap val) {
+        this(val, null);
+    }
+
+    public ExtWrapArrayListMultiMap(Multimap val, ExternalizableWrapper keyType) {
+        if (val == null) {
+            throw new NullPointerException();
+        }
+
+        this.val = val;
+        this.keyType = keyType;
+    }
+
+    /* Constructors for deserialization */
+
+    public ExtWrapArrayListMultiMap() {
+    }
+
+
+    public ExtWrapArrayListMultiMap(Class keyType) {
+        this(new ExtWrapBase(keyType));
+    }
+
+    public ExtWrapArrayListMultiMap(ExternalizableWrapper keyType) {
+        if (keyType == null) {
+            throw new NullPointerException();
+        }
+
+        this.keyType = keyType;
+    }
+
+    @Override
+    public ExternalizableWrapper clone(Object val) {
+        return new ExtWrapArrayListMultiMap((Multimap)val, keyType);
```

Comment:
I'm probably misremembering, but should this method call deep-copy the `val` object? (Easiest to just check other implementations)

Reply:
Thanks for the prompt. I checked other implementations ([here](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/javarosa/core/util/externalizable/ExtWrapMap.java#L76) and [here](https://github.com/dimagi/commcare-core/blob/master/src/main/java/org/javarosa/core/util/externalizable/ExtWrapList.java#L65)) and it seems to be a shallow copy.